llama-index-core/llama_index/core/async_utils.py

```diff
@@ -101,9 +105,11 @@ async def run_jobs(
         List[Any]:
             List of results.
     """
+    parent_span_id = dispatcher.current_span_id
     asyncio_mod = get_asyncio_module(show_progress=show_progress)
     semaphore = asyncio.Semaphore(workers)
 
+    @dispatcher.async_span_with_parent_id(parent_id=parent_span_id)
```

Comment:
So basically, anything span that spawns many jobs should use this decorator right?

Reply:
Yes with `async` that's right. The outer function to this should be decorated and we can extract the `parent_id`.

In fact, to be safe, we should use `run_jobs()` whenever we want to instrument the "simultaneous" execution of async tasks.